core/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -39,4 +39,16 @@ public static void main(String[] args) throws StartupError {
             throw new StartupError(t);
         }
     }
+
+    /**
+     * Required method that's called by Apache Commons procrun when
+     * running as a service on Windows, when the service is stopped.
+     *
+     * http://commons.apache.org/proper/commons-daemon/procrun.html
+     *
+     * NOTE: If this method is renamed and/or moved, make sure to update service.bat!
+     */
+    public static void close(String[] args) {
```

Comment:
Must this really be public? According to the documentation it need only be package private. That would be greatly preferable, because it makes it easier to protect down the road.